pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java

```diff
@@ -360,6 +359,9 @@ public ServerType getServerType() {
     public enum SegmentType {
       OFFLINE, REALTIME
     }
+
+    @Deprecated
+    public static final String TABLE_NAME = "segment.table.name";
```

Comment:
we still need it for SegmentMetadataImpl right? We are deprecating only from zk metadata?

Reply:
The only current usage of table name from `SegmentMetadataImpl` is when deriving a table name from segment metadata during segment upload. Eventually, I would like to replace this with table name header from upload.

The reason why I want to put deprecated annotation now is to prevent people from using it anywhere else. How do you think?

Reply:
In that case, this call should not be deprecated now. It can be deprecated when we have the alternative way of determining table name (or, still kept in there to validate). Not sure if header is the best way, or a query parameter, or some sort of signed object that indicates the table name, and other data.

Reply:
@mcvsubbu I double checked. `CommonConstants.Segment.TABLE_NAME` is used for `segment zk metadata`. Segment metadata is using `V1Constants.Segment.TABLE_NAME`. I reverted deprecate annotation for all segment metadata related one and only added deprecated annotation for  segment zk metadata.

I accidentally ran commit ammend that the diff is gone :( Would you review this once more?